hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AEndpointRegion.java

```diff
@@ -257,6 +275,44 @@ public void testWithVPCE() throws Throwable {
     expectInterceptorException(client);
   }
 
+  @Test
```

Comment:
we should also add a test for a key that  does not exist, eg:

```
newFS.getObjectMetadata("nonExistentKey.txt");
```
verify that we get a 404 and not a 400.

Reply:
Yes, assertion with `newFS.exists(srcFilePath)` covers it internally, that should be good?